%TODO: How much to describe each feature? refer to appendix? what if not in appendix?

\chapter{Prior Work In Interface Prediction}
\label{chap:relatedwork} 

%In silico prediction of interfaces began in the 

As previously mentioned, the experimental determination of protein complexes is time and resource intensive, prompting computational modeling approaches.
Esmaielbeiki et al describe three slightly different objectives in this vein, protein interaction prediction, protein-protein docking, and protein interfaces prediction~\cite{esmaielbeiki2015}.
The first objective seeks to identify pairs of proteins which interact, elucidating the complex protein interaction networks that give rise to cellular processes. 
The second objective considers two specified proteins and seeks the bound 3D structure of their complex.
The third objective, and the focus of this thesis, is chiefly concerned with identifying the specific residues or pairs of residues which make up the interface between two proteins.
It is less concerned with the 3D complex structure as with the interface itself.

It is worth noting that docking methods can also be used to predict interfaces, by first solving for the 3D structure of the complex and then extracting the interface from the complex.
Indeed, docking methods were some of the earliest computational approaches developed to model protein interactions~\cite{janin1995}.
Docking begins with the known unbound structures of two proteins known to interact, and conducts two main steps: search and scoring.
During search, two proteins are translated and rotated relative to each other and brought into contact to create a putative 3D bound structure for the complex.
The putative structures are then evaluated by a scoring function to identify the structures that are most likely to be part of the complex.
Docking methods differ in their the search algorithm and scoring function.
Scoring functions may incorporate complementarity in geometry, chemistry, and electrostatics, or incorporate van der Walls forces or evidence based (aka statistical) potentials~\cite{tuncbag2011}\cite{janin1995}.
One of the major advantages of docking is its ability to produce interface predictions \textit{ab initio} without requiring examples of known interfaces, which is particularly useful when experimental complex data are sparse.
Unfortunately, docking methods traditionally suffer from relatively high false positive rates, are considerably less effective for complexes which undergo conformational change when binding, and are computationally expensive because of the vast search space~\cite{janin1995}\cite{tuncbag2011}.

Some early alternatives to docking also avoided using example 3D bound structures, instead relying on sequence information, residue properties, and unbound structures for each protein in the complex.
Lichtarge et al~\cite{lichtarge1996} used inferred evolutionary relationships between different proteins to identify conserved residues and then identified those conserved residues which lay on the surface of the protein.
This method was based on the hypothesis that conserved surface residues must be vital to a protein's function and therefore probably constitute an interface.
Pazos et al~\cite{pazos1997} take a similar approach, but instead look at evolutionary relationships between protein complexes and identify pairs of residues between the proteins in the complex which have co-evolved.
This method requires only sequence information and therefore is applicable even in cases where the protein structures are unknown. 
Additionally, this method is partner specific since it identifies residue pairs which show correlated changes.
Gallet et al~\cite{gallet2000} use a sliding window on a protein sequence and calculate measures of hydrophobicity in a region, which can easily be calculated knowing the residue identities and secondary structure.
This method requires no phylogenetic information so is applicable even when no close evolutionary relatives can be identified.

Early methods such as those listed above were crafted for the available data and computational resources of the time, but were unable to fully account for the growing body of research surrounding protein interfaces and their properties as in (Jones \& Thornton, 1996)~\cite{jones1996}.
It was (Jones \& Thornton, 1997) that proposed a method which incorporated multiple structural features such as surface planarity, planarity, and accessible surface area, with residue level features such as solvation potential, hydrophobicity, and interface residue propensity.
They constructed a manual scoring function which inputs were the aforementioned features and output was a score, where higher scores corresponded with members of the interface.
They constructed a different scoring function for each of three different categories of complex, reflecting observations made into the characteristics of different complex types.
Prediction was performed on patches of residues.

Evaluation of early methods was challenging due to the paucity of experimentally determined protein interfaces~\cite{esmaielbeiki2015}.
Thankfully, the turn of the twenty first century coincided with an increase in the number of experimentally determined structures added to only databases such as the Protein Data Bank~\cite{berman2000}
Curated subsets also emerged which focused on evaluating protein-protein docking methods, such as the Critical Assessment of Predicted Interactions (CAPRI)~\cite{janin2003} and the Docking Benchmark Dataset (DBD)~\cite{chen2002}.
These subsets also became useful in the evaluation (and sometimes training) of interface detection methods.
	
This increasing availability of data and increased interest in the problem led to a growing number and diversity of approaches.
The emergent of class known as \textit{template based} methods utilized a non-redundant library of known protein interfaces to make predictions about unforeseen proteins.
For a given query protein, a search is made in the library for known complexes where one partner is similar to the query protein.
The interface of the query protein is then inferred from the interfaces of the most similar query results.
Similarity may be measured via homology (sequence similarity) or structural similarity~\cite{esmaielbeiki2015}.

Other data-based methods have appeared which are based on either machine learning or statistical methods.
Some early machine learning based approaches used a support vector machine (SVM) to classify residues as either belonging to an interface or not.
%TODO: explain SVMs?
An SVM essentially provides a scoring function which is dependent on training data rather than manually constructed.
(Koike \& Takagi, 2004~\cite{koike2004}) trained an SVM classifier to perform partner independent prediction of interfaces.
They represent a residue by its profile, a vector of relative abundances of each amino acid type among homologous proteins at that location (see Appendix \ref{appendix:features} for more detail).
They experiment with different feature representations to make predictions at a particular residue, finding that incorporating profiles from sequential or spatially neighboring residues improves performance, as does incorporating accessible surface area and accounting for the relative interface size.
(Bradford \& Westhead, 2004~\cite{bradford2004}) also perform perform partner independent prediction with a SVM classifier, but instead make predictions for surface patches rather than individual residues.
(Zhou \& Shan, 2001~\cite{zhou2001}) were among the first to train a neural network for partner independent prediction.
They incorporated profile and solvent exposure of residues and their neighbors to make predictions at the residue level.
Their method uses residue profiles
In a follow up paper, (Chen \& Zhou, 2005~\cite{chen2005}) use an ensemble of neural networks to make a consensus prediction concerning a residue of interest.
%TODO: talk about proto-convolution of chen2005?

		naive bayesian
		hmm
		conditional random field
		
		
	typically incorporate both sequence and structure information and information from sequential/structural neighbors
	not global solutions like docking
	still partner independent
	
zhou and qin identify need for partner specific methods to increase specificity of predictions (zhou 2007)
		



\section{Latest Partner Specific Methods}
PPIPP
PairPRED



Interfaces exhibit a local spatial correlation in that a residue is more likely to be a part of an interface if nearby residues are also a part of the interface. 
Local spatial correlation can aid prediction if information from nearby residues is taken into account during prediction.


Such methods exist, but are often based on energy minimization techniques which are computationally expensive and attempt to model the entire 3D structure of the interface at once~\cite{esmaielbeiki2015}.
Some of these methods (Haddock~\cite{zundert2016} for example) allow the user to suggest amino acid residues which are likely part of the interface, to help bias the algorithm towards a correct solution.
This has motivated work to predict amino acid pairs which constitute part of the interface without solving for the entire interface at once. 




